### PR TITLE
fix: add permission checks for creating maps and shared files

### PR DESCRIPTION
**Pull request by paulschreiber**
## Description
Add permission checks for creating maps and shared files

### Related Issues
Fixes https://github.com/techmatters/terraso-web-client/issues/811
Web client https://github.com/techmatters/terraso-web-client/pull/859

### CODE REVIEW

**Review by shrouxm — APPROVED** [review]
LGTMy inexpert eyes

File: terraso_backend/apps/core/models/groups.py [hunk 1]
```diff
@@ -153,6 +153,9 @@ def managers_only(self):
     def approved_only(self):
         return self.filter(membership_status=Membership.APPROVED)
 
+    def approved_only(self):
+        return self.filter(membership_status=Membership.APPROVED)
+
```
Comment (david-code) [inline terraso_backend/apps/core/models/groups.py:158]:
```suggestion
```

Seems to be duplicated from above code

File: terraso_backend/apps/shared_data/permission_rules.py [hunk 2]
```diff
@@ -29,6 +29,11 @@ def allowed_to_delete_data_entry(user, data_entry):
     )
 
 
+@rules.predicate
+def allowed_to_add_data_entry(user, group):
+    return user.memberships.approved_only().filter(group__in=[group]).exists()
```
Comment (david-code) [inline terraso_backend/apps/shared_data/permission_rules.py:34]:
```suggestion
    return user.memberships.approved_only().filter(group=group).exists()
```
It seems that the list is redundant here.

**Review by david-code — APPROVED** [review]
Looks good!

File: terraso_backend/apps/core/models/groups.py (hunk 1 above)
Comment (paulschreiber) [reply terraso_backend/apps/core/models/groups.py]: Weird! Must have been duplicated in a rebase.

**Pull request merged**